canonical-web-and-design/maas-ui
# feat(ui): Build RSDDetails components in React

## Done

- Built RSDDetails parent components.
- Renamed KVMDetails components to PodDetails for those that were almost 100% identical to what was needed for RSDDetails. 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /r/rsd and click on an RSD
- Check that the RSD details page loads and the data is displayed correctly. Note that there will never be a View by NUMA node option in this case, or hugepage data, and the copyable code is for Power address, not LXD URL or Virsh.
- Check that you can submit the configuration form.

## Fixes

Fixes #1639 

## Screenshot
![0 0 0 0_8400_MAAS_r_rsd_30](https://user-images.githubusercontent.com/25733845/93842451-58bbe380-fcda-11ea-8a89-247f03624b7f.png)
